### PR TITLE
feat(login google) : Google Login 

### PR DESCRIPTION
**Pull request by unadine**
## What does this PR do?
Login to barefoot with google
## Description of Task to be completed?
As a user, I can login in to Barefoot application via  Google.

## How should this be manually tested?
npm run test

### CODE REVIEW

File: src/database/models/user.js [hunk 1]
```diff
@@ -6,11 +7,10 @@ module.exports = (sequelize, DataTypes) => {
   /** Class representing a User model . */
   class User extends Model {
     /**
-* @description this method to associate trip model to user model
-* @param {Object} models
-* @returns {object} returns associations
-* @memberof User
-*/
+     * Helper method for defining associations.
```
Comment (James-nyagatare) [inline src/database/models/user.js]: can you please use valid jsdocs

File: src/index.js [hunk 2]
```diff
@@ -31,6 +32,8 @@ app.use('/public', express.static(path.join(__dirname, '../public')));
 
 routes(app);
 
+// app.use('/api/v1', routes);
```
Comment (James-nyagatare) [inline src/index.js]: can you please remove this unnecessary comment

**Review by James-nyagatare — CHANGES_REQUESTED** [review]
Welldone @unadine, can you please consinder implementing the feedbacks I left on this PR.

File: src/database/models/user.js (hunk 1 above)
Comment (unadine) [reply src/database/models/user.js]: done

File: src/index.js (hunk 2 above)
Comment (unadine) [reply src/index.js]: done

**Comment by harerakalex** [comment]
@unadine kindly fix the failing tests.

File: src/controllers/socialAuthController.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+   * @return {object} Oject of data or error
+  */
+  static async successSignUp(req, res) {
+    try {
+      const { user: { email } } = req;
+      const token = generateToken(await UserService.findUser({ email }));
+      return Response.success(res, 200, 'Successfully logged in', token);
+    } catch (error) {
+      return Response.error(res, 500, error);
+    }
```
Comment (Mucyo-Aime-Christian) [inline src/controllers/socialAuthController.js]: would you mind if you use the status code file instead of writing 200 or 500 please.

**Review by Mucyo-Aime-Christian — COMMENTED** [review]
well done @unadine, however, work on the below comments.

**Review by Mucyo-Aime-Christian — APPROVED** [review]
LGTM

**Pull request merged**